lib/runtime-c-api/src/lib.rs

```diff
@@ -1436,14 +1421,12 @@ impl From<Type> for wasmer_value_tag {
 
 impl From<wasmer_value_tag> for Type {
     fn from(v: wasmer_value_tag) -> Self {
-        unsafe {
-            match v {
-                wasmer_value_tag::WASM_I32 => Type::I32,
-                wasmer_value_tag::WASM_I64 => Type::I64,
-                wasmer_value_tag::WASM_F32 => Type::F32,
-                wasmer_value_tag::WASM_F64 => Type::F64,
-                _ => panic!("not implemented"),
```

Comment:
If I get it correctly, the `_` arm is unreachable, but we don't trust the `v` input, is that right? In other words, we don't assume that `v` is correctly built.